File: 2025/day_1/day1a.js
// https://adventofcode.com/2025/day/1
const fs = require('fs');
const input = fs.readFileSync('./input1.txt', 'utf8');

const turns = input.split('\n').map(v => (v[0] === "L" ? -1 : 1) * (parseInt(v.slice(1)) % 100));

let value = 50, zeros = 0;
turns.forEach((v) => {
	value += v;
	if (value < 0) value += 100;
	if (value > 99) value -= 100;
	if (value === 0) zeros++;
});
console.log(zeros);
// 1195

File: 2025/day_1/day1a.py
# https://adventofcode.com/2025/day/1

# load input file
with open("input1.txt", "r") as file:
	lines = file.read().splitlines()

turns = [
	(-1 if line[0] == "L" else 1) * (int(line[1:]) % 100) 
	for line in lines
]

value, zeros = 50, 0
for v in turns:
	value = (value + v) % 100
	if value == 0: zeros += 1

print(zeros)
# 1195
